main.go

```diff
@@ -423,15 +459,52 @@ func main() { //nolint:funlen,maintidx
 	}
 }
 
-func createSecretCacheConfig(platform cluster.Platform) map[string]cache.Config {
+func createSecretCacheConfig(ctx context.Context, cli client.Client, platform cluster.Platform) (map[string]cache.Config, error) {
 	namespaceConfigs := map[string]cache.Config{
-		"istio-system":      {}, // for both knative-serving-cert and default-modelregistry-cert,as an easy workarond, to watch all in this namespace for now
+		"istio-system":      {}, // for both knative-serving-cert and default-modelregistry-cert, as an easy workarond, to watch both in this namespace
 		"openshift-ingress": {},
 	}
+	// TODO: if we dont want harcoded above two namespace we can add them with label selector
+	// maistra.io/member-of=istio-system
+	// network.openshift.io/policy-group=ingress
+
+	cNamespaceList := &corev1.NamespaceList{}
+
+	// if user create namespace and want it to be used as application namespace
+	// they need to create label "opendatahub.io/application-namespace": "true" then set DSCI to use it
+	// we only support only namespace in the cluster has this label
+	labelSelector := client.MatchingLabels{
+		labels.CustomizedAppNamespace: labels.True,
+	}
+
+	if err := cli.List(ctx, cNamespaceList, labelSelector); err != nil {
+		return map[string]cache.Config{}, err
+	}
+	if len(cNamespaceList.Items) > 1 {
+		return map[string]cache.Config{}, errors.New("found multiple namespace with label: opendatahub.io/application-namespace: true")
+	}
+	for _, ns := range cNamespaceList.Items {
+		namespaceConfigs[ns.Name] = cache.Config{}
+	}
+
+	// get all default namespaces by label.
+	namespaceList := &corev1.NamespaceList{}
+	labelSelector = client.MatchingLabels{
+		labels.ODH.OwnedNamespace: labels.True,
+	}
+	if err := cli.List(ctx, namespaceList, labelSelector); err != nil {
+		return map[string]cache.Config{}, err
+	}
+
+	for _, ns := range namespaceList.Items {
+		namespaceConfigs[ns.Name] = cache.Config{}
```

Comment:
I would expect that we only handle one of the two cases, here it would be possible that we creates two entries in the cache, but we should only have one